### PR TITLE
Upgrade all packages with "flutter update-packages --force-upgrade"

### LINKED ISSUE #18195: testWidgets causes incorrect file/line/column for tests (returns location of the test() call inside flutter_test)
The test framework returns the location for a test so that an editor can jump to it when the user clicks on it. For tests that use `testWidgets` it returns the location of the `test` call inside `testWidgets` so it won't take the user to where they expect.

Example:

```js
{
	"test": {
		"id": 3,
		"name": "Changing locale",
		"suiteID": 0,
		"groupIDs": [
			2
		],
		"line": 60,
		"column": 18,
		"url": "package:flutter_test/src/widget_tester.dart"
	},
	"type": "testStart",
	"time": 21952
}
```

I need to use this information to navigate to the test when a user clicks in the tree on the left. Currently I ignore any `package:` URIs so it just does nothing, but it makes the test runner much less useful.

screen shot 2018-06-05 at 10 42 33 am

### PR DESCRIPTION
**Pull request by DanTup**
I'm updating `test` to fix #18195.

~~Running `flutter update-packages --force-upgrade` updates everything which seems kinda risky so I've only updated the test package manually. The `flutter_test` package tests all pass locally, but I'm opening the PR to ensure a full CI run (so I'll mark this as do-not-review until that all looks good).~~

### CODE REVIEW

**Comment by DanTup** [comment]
@devoncarew Turns out this wasn't safe - `test` had a bunch of deps updated/changed between .37 and .40 so this fails with ots of:

> Warning: pubspec in /tmp/flutter sdk/dev/manual_tests has invalid dependencies. Please run "flutter update-packages" --force-upgrade to update them correctly.

So, I'm gonna update this PR to just have everything updated to see how it builds, but I have no idea if rolling the dependencies on a whim to get a test fix is a good idea!

**Comment by DanTup** [comment]
Looks like this upgrade brought a breaking change:

> error • The getter 'data' isn't defined for the class 'Image' • dev/integration_tests/ui/test_driver/screenshot_test.dart:35:30

I'm not sure if this is expected (I don't know if the command I ran was supposed to only take compatible updates?) so I'll wait for some input on how to proceed before I start fixing things.

File: dev/devicelab/pubspec.yaml [hunk 1]
```diff
@@ -8,67 +8,66 @@ environment:
 
 dependencies:
   args: 1.4.3
-  file: 5.0.0
-  image: 1.1.32
+  file: 5.0.1
+  image: 1.1.14
```
Comment (xster) [inline dev/devicelab/pubspec.yaml]: I tried doing this over the weekend too but it ended up downgrading this which causes some issues.

Comment (DanTup) [reply dev/devicelab/pubspec.yaml]: Good spot, I didn't realise it had gone down. Maybe I'll try setting that one back manually and see what happens.

**Comment by DanTup** [comment]
@xster here's the reason it's giving for downgrading image:

```
Because image >=1.1.15 depends on archive >=1.0.16 <1.1.0 and flutter_devicelab depends on archive 2.0.0, image >=1.1.15 is forbidden.
So, because flutter_devicelab depends on image 1.1.32, version solving failed.
```

I found this by changing the pubspec in devicelab back to 1.1.32 and then running pub get.

**Comment by DanTup** [comment]
So, running this upgrade has upgraded `archive` to `2.0.0` but `image >= 1.1.15` are pinned to `archive <= 1.1.0` (and have been for 4 years looking at GitHub).

So the problem here is that upgrading `archive` to v2.0 is blocked by `image`..

**Comment by DanTup** [comment]
I've opened https://github.com/flutter/flutter/issues/18402 for discussing the issue I hit. I'm aborting this one for now and will re-visit when I know what's going on.

**Pull request closed without merging**